server/grpc/server.go

```diff
@@ -69,6 +71,13 @@ func (s *Server) Close() {
 	s.instance.GracefulStop()
 }
 
+func validateInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
```

Comment:
I'm not sure it's the best place here.
I would like the SDK to also check its inputs and a lot of grpc request message are directly forward to the sdk.
maybe check at both place?

Reply:
Why SDK should check that. For now we use only grpc as entry for sdk. SDK transforms over time and for now, you just can't use it without running engine. So it should stay here for now. And when sdk transforms to the point where you can use it without engine, then we might add the validation here.

Reply:
I just think that the verification should be done in last entry point of the app.
We also plan to move to SDK the definition of the message, so it will make even more sense to already put the verification in SDK.

Reply:
what is the last entry point? For now, it is the cosmos app and there messages are validated. SDK is a kind of middleware between our grpc api and cosmos api. 

> We also plan to move to SDK the definition of the message, 

Then we will move the validation as well, but for now let's keep it as it is.